app/code/Magento/Sales/Controller/Adminhtml/Order/CreditmemoLoader.php

```diff
@@ -181,7 +188,13 @@ public function load()
         $creditmemoId = $this->getCreditmemoId();
         $orderId = $this->getOrderId();
         if ($creditmemoId) {
-            $creditmemo = $this->creditmemoRepository->get($creditmemoId);
+            try {
+                $creditmemo = $this->creditmemoRepository->get($creditmemoId);
+            } catch (NoSuchEntityException $e) {
+            } catch (InputException $e) {
+                $this->messageManager->addErrorMessage(__('This credit memo no longer exists.'));
```

Comment:
I think it will be good to use the same message for `NoSuchEntityException` case